d4s/src/main/scala/d4s/DynamoComponent.scala

```diff
@@ -21,14 +21,14 @@ object DynamoComponent {
 
   final class Impl[F[+_, +_]: BIO](
     conf: DynamoConfig,
-    portCheck: PortCheck @Id("dynamo-port")
+    portCheck: PortCheck @Id("dynamo-port"),
   ) extends DIResource[F[Throwable, ?], DynamoComponent]
     with IntegrationCheck {
 
     override def resourcesAvailable(): ResourceCheck = {
       conf.maybeLocalUrl.fold(ResourceCheck.Success(): ResourceCheck) {
         url =>
-          portCheck.checkUri(URI.create(url), 8042, "DynamoClient")
+          portCheck.checkUrl(new URL(url), "DynamoClient")
```

Comment:
@neko-kai Url instead of port, since we have complete url here without any ports. This will allow users to use their own ports during development.